### PR TITLE
Gamedb: fix up some game titles

### PR DESCRIPTION
**Pull request by Mrlinkwii**
## Rationale behind Changes

fixeup some wrong titles 
### Suggested Testing Steps

make sure CI passes

### CODE REVIEW

File: bin/resources/GameIndex.yaml [hunk 1]
```diff
@@ -18578,7 +18578,7 @@ SLES-54705:
   name: "SBK-07 - Superbike World Championship"
   region: "PAL-M5"
 SLES-54708:
-  name: "The History Channel - Great Battles of Rome"
+  name: "History Channel - Great Battles of Rome, The"
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:18581]:
```suggestion
  name: "History Channel, The - Great Battles of Rome"
```

File: bin/resources/GameIndex.yaml [hunk 2]
```diff
@@ -19287,7 +19287,7 @@ SLES-54983:
   name: "Postman Pat"
   region: "PAL-M11"
 SLES-54984:
-  name: "The Water Horse - Legend of the Deep"
+  name: "Water Horse - Legend of the Deep, The"
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:19290]:
```suggestion
  name: "Water Horse, The - Legend of the Deep"
```

File: bin/resources/GameIndex.yaml [hunk 3]
```diff
@@ -19638,22 +19638,22 @@ SLES-55138:
...
   region: "PAL-E"
 SLES-55143:
-  name: "The Chronicles of Narnia - Prince Caspian"
+  name: "Chronicles of Narnia - Prince Caspian, The"
   region: "PAL-G-I"
   gsHWFixes:
     halfPixelOffset: 2 # Reduces ghosting but on normal vertex is less ghosting but causes corruption at the sides.
 SLES-55144:
-  name: "The Chronicles of Narnia - Prince Caspian"
+  name: "Chronicles of Narnia - Prince Caspian, The"
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:19646]:
```suggestion
  name: "Chronicles of Narnia, The - Prince Caspian"
```

**Review by RedDevilus — CHANGES_REQUESTED** [review]
+ Others to fit correctly. The should be together with the words

File: bin/resources/GameIndex.yaml (hunk 1 above)
Comment (refractionpcsx2) [reply bin/resources/GameIndex.yaml:18581]: Still incorrect

File: bin/resources/GameIndex.yaml (hunk 2 above)
Comment (refractionpcsx2) [reply bin/resources/GameIndex.yaml:19290]: Still incorrect

File: bin/resources/GameIndex.yaml [hunk 4]
```diff
@@ -19563,7 +19563,7 @@ SLES-55091:
   name: "Hoppie"
   region: "PAL-E"
 SLES-55102:
-  name: "The History Channel - Battle for the Pacific"
+  name: "History Channel -The Battle for the Pacific"
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:19566]: , The before the -

File: bin/resources/GameIndex.yaml [hunk 5]
```diff
@@ -19638,22 +19638,22 @@ SLES-55138:
...
     halfPixelOffset: 2 # Reduces ghosting but on normal vertex is less ghosting but causes corruption at the sides.
 SLES-55145:
-  name: "The Chronicles of Narnia - Prince Caspian"
+  name: "Chronicles of Narnia, The - Prince Caspian"
   region: "PAL-F-DU"
   gsHWFixes:
     halfPixelOffset: 2 # Reduces ghosting but on normal vertex is less ghosting but causes corruption at the sides.
 SLES-55146:
-  name: "The Chronicles of Narnia - Prince Caspian"
+  name: "Chronicles of Narnia, The -  Prince Caspian"
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:19656]:
```suggestion
  name: "Chronicles of Narnia, The - Prince Caspian"
```

File: bin/resources/GameIndex.yaml [hunk 6]
```diff
@@ -23137,7 +23137,7 @@ SLPM-62345:
   name: "Simple 2000 Series Vol. 32 - The Sensha"
   region: "NTSC-J"
 SLPM-62348:
-  name: "The Mechsmith - Rum"
+  name: "Mechsmith,The - Rum"
```
Comment (RedDevilus) [inline bin/resources/GameIndex.yaml:23140]:
```suggestion
  name: "Mechsmith, The - Rum"
```

**Review by RedDevilus — CHANGES_REQUESTED** [review]
Minor mistakes still

File: bin/resources/GameIndex.yaml [hunk 7]
```diff
@@ -35833,7 +35833,7 @@ SLPS-25934:
   name: "Samurai Spirits - Rokuban Shoubu [NeoGeo Online Collection the Best]"
   region: "NTSC-J"
 SLPS-25935:
-  name: "The King of Fighters '98 Ultimate Match (NeoGeo Online Collection the Best)"
+  name: "King of Fighters '98 Ultimate Match (NeoGeo Online Collection the Best), The"
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:35836]:
```suggestion
  name: "King of Fighters '98 Ultimate Match, The (NeoGeo Online Collection the Best)"
```

Comment (refractionpcsx2) [reply bin/resources/GameIndex.yaml:35836]: I will say however how these are done feels somewhat inconsistent, but there's no clear solution, but the The should certainly come before the packaging type.

**Pull request merged**